hazelcast-documentation/src/RingBuffer.md

```diff
@@ -1,47 +1,233 @@
-  E readOne(long sequence) throws InterruptedException;
-  ICompletableFuture<Long> addAllAsync(Collection<? extends E> collection, 
-                        OverflowPolicy overflowPolicy);
-  ICompletableFuture<ReadResultSet<E>> readManyAsync(long startSequence, 
-                         int minCount, int maxCount, 
-                         IFunction<E, Boolean> filter);
+### Synchronous and Asynchronous backups.
+A Ringbuffer by default has a single synchronous backup. But this can be controlled, just like most of the other Hazelcast 
+distributed data-structures by setting the sync and async backups. In the example below a ringbuffer is configured with 0
+sync backups and 1 async backup:
+
+```
+<ringbuffer name="rb">
+    <backup-count>0</backup-count>
+    <async-backup-count>1</async-backup-count>
+</ringbuffer>
 ```
+An async backup will probably give you better performance, but there is a chance that the chance that the item added is lost 
+when the member owning the primary crashed before the replication has completed. Probably you want to have a look at the batching
+methods if you need high performance but don't want to give up on consistency.
 
+### Time to live.
+The RingBuffer can be configured with a time to live in seconds. Using this setting, you can control how long the items remain in 
+the RingBuffer before they are expired. By default, the time to live is set to 0, meaning that unless the item is overwritten, 
+it will remain in the RingBuffer indefinitely. If a time to live is set and an item is added, then depending on the OverwritePolicy, 
+either the oldest item is overwritten, or the call is rejected. 
 
-The RingBuffer can be configured with a time to live in seconds. Using this setting, you can control how long the items remain in the RingBuffer before they are deleted. By default, the time to live is set to 0, meaning that unless the item is overwritten, it will remain in the RingBuffer indefinitely. If a time to live is set and an item is added, then depending on the OverwritePolicy, either the oldest item is overwritten, or the call is rejected. 
+In the code fragment below a ringbuffer is configured with a time to live of 180 seconds.
+```
+<ringbuffer name="rb">
+    <time-to-live-seconds>180</time-to-live-seconds>
+</ringbuffer>
+```
 
-The RingBuffer can also be configured with an InMemoryFormat which controls the format of stored items. By default, `BINARY` is used, meaning that the object is stored in a serialized form. You can select the `OBJECT` InMemoryFormat, which is useful when filtering is applied or when the `OBJECT` InMemoryFormat can lead to a smaller memory footprint than a `BINARY`. 
+### Overflow policy
+using the overflow policy one can determine what to do if the oldest item in the ringbuffer is not old enough to expire. There
+are currently 2 options available:
+* OverflowPolicy.OVERWRITE: in this case the oldest item is overwritten. You might wonder why not set the time to live to 0, since
```

Comment:
I don't understand that point - non of the sentences. Sounds like you got some changes into that don't fit together :)